homeassistant/components/derivative/config_flow.py

```diff
@@ -31,50 +30,54 @@
 )
 
 UNIT_PREFIXES = [
-    {"value": "none", "label": "none"},
-    {"value": "n", "label": "n (nano)"},
-    {"value": "µ", "label": "µ (micro)"},
-    {"value": "m", "label": "m (milli)"},
-    {"value": "k", "label": "k (kilo)"},
-    {"value": "M", "label": "M (mega)"},
-    {"value": "G", "label": "G (giga)"},
-    {"value": "T", "label": "T (tera)"},
-    {"value": "P", "label": "P (peta)"},
+    selector.SelectOptionDict(value="none", label="none"),
+    selector.SelectOptionDict(value="n", label="n (nano)"),
+    selector.SelectOptionDict(value="µ", label="µ (micro)"),
+    selector.SelectOptionDict(value="m", label="m (milli)"),
+    selector.SelectOptionDict(value="k", label="k (kilo)"),
+    selector.SelectOptionDict(value="M", label="M (mega)"),
+    selector.SelectOptionDict(value="G", label="G (giga)"),
+    selector.SelectOptionDict(value="T", label="T (tera)"),
+    selector.SelectOptionDict(value="P", label="P (peta)"),
 ]
 TIME_UNITS = [
-    {"value": TIME_SECONDS, "label": "Seconds"},
-    {"value": TIME_MINUTES, "label": "Minutes"},
-    {"value": TIME_HOURS, "label": "Hours"},
-    {"value": TIME_DAYS, "label": "Days"},
+    selector.SelectOptionDict(value=TIME_SECONDS, label="Seconds"),
+    selector.SelectOptionDict(value=TIME_MINUTES, label="Minutes"),
+    selector.SelectOptionDict(value=TIME_HOURS, label="Hours"),
+    selector.SelectOptionDict(value=TIME_DAYS, label="Days"),
 ]
 
 OPTIONS_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_ROUND_DIGITS, default=2): selector.selector(
-            {
-                "number": {
-                    "min": 0,
-                    "max": 6,
-                    "mode": "box",
-                    CONF_UNIT_OF_MEASUREMENT: "decimals",
-                }
-            }
+            selector.SelectorType.NUMBER,
+            selector.NumberSelectorConfig(
+                min=0,
+                max=6,
+                mode=selector.NumberSelectorMode.BOX,
+                unit_of_measurement="decimals",
+            ),
+        ),
+        vol.Required(CONF_TIME_WINDOW): selector.selector(
+            selector.SelectorType.DURATION
         ),
-        vol.Required(CONF_TIME_WINDOW): selector.selector({"duration": {}}),
         vol.Required(CONF_UNIT_PREFIX, default="none"): selector.selector(
-            {"select": {"options": UNIT_PREFIXES}}
+            selector.SelectorType.SELECT,
+            selector.SelectSelectorConfig(options=UNIT_PREFIXES),
         ),
```

Comment:
Was just talking with @emontnemery, should this become just this:

```py
        vol.Required(CONF_UNIT_PREFIX, default="none"): selector.SelectSelector(
            selector.SelectSelectorConfig(options=UNIT_PREFIXES),
        ),
```

Reply:
Yeah, either that, or drop the `SELECTORS` registry.
If we add support for validating a selector in a blueprint, we will need to check that the selector is either a member of the `StrEnum` or of the `SELECTORS` registry, is one preferred over the other?

Reply:
If we check if "a member", then it still needs to have a mapping to map the enum member to the selector class. I think the registry might still be useful for just the "Here is a bunch of config from the frontend" (`selector.selector()`) case.

So I guess that makes the StrEnum maybe obsolete/unneeded in this case.

Reply:
I like this approach, it makes it not a breaking change. Unfortunately we lose the typehints for the right `Config` `TypedDict` class to use for a given selector but I think that's OK given the naming conventions we use

Reply:
nvm, I added the necessary things to get typing still